include/ginkgo/core/solver/multigrid.hpp

```diff
@@ -93,6 +98,70 @@ enum class cycle { v, f, w, kfcg, kgcr };
 enum class mid_smooth_type { both, post_smoother, pre_smoother, standalone };
 
 
+struct MultigridState {
```

Comment:
This is needed only to persist between apply calls right, and is mostly an implementation detail?

Reply:
yes. it contains the data and implementation